components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -2786,4 +2819,95 @@ void RewardsServiceImpl::OnDeleteActivityInfo(
   }
 }
 
+void RewardsServiceImpl::GetAutoContributeCount(
+    rewards_counter::OnDataCountedCallback callback) {
+  bat_ledger_->GetAutoContributeCount(
+      base::BindOnce(&RewardsServiceImpl::OnGetAutoContributeCount,
+      AsWeakPtr(), std::move(callback)));
+}
+
+void RewardsServiceImpl::OnGetAutoContributeCount(
+    rewards_counter::OnDataCountedCallback callback,
+    int64_t count,
+    uint64_t previous_reconcile_stamp) {
+  base::string16 previous_reconcile_date;
+  if (previous_reconcile_stamp != 0) {
+    previous_reconcile_date = TimeFormatShortDate(
+        base::Time::FromDoubleT(previous_reconcile_stamp));
+  }
+  std::move(callback).Run(count, previous_reconcile_date);
+}
+
+void RewardsServiceImpl::RemoveData(
+    int remove_mask,
+    OnRemoveDataCallback callback) {
+  if (!Connected()) {
+    return;
+  }
+  bat_ledger_->RemoveData(remove_mask,
+    base::BindOnce(&RewardsServiceImpl::OnRemoveData,
+      AsWeakPtr(), std::move(callback)));
+}
+
+void RewardsServiceImpl::OnRemoveData(
+    OnRemoveDataCallback callback,
+    bool result) {
+  for (auto& observer : observers_) {
```

Comment:
I would make the callback before calling the observers

Reply:
Done.